test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1

```diff
@@ -99,19 +136,48 @@ try
 
     $coverallsExe = Join-Path $coverallsPath "tools\csmacnz.Coveralls.exe"
     $coverallsParams = @("--opencover",
-                        "-i $outputLog",
-                        "--repoToken $coverallsToken",
-                        "--commitId $commitId",
-                        "--commitBranch master",
-                        "--commitAuthor `"$author`"",
-                        "--commitEmail $email",
-                        "--commitMessage `"$message`""
-                        )
+        "-i $outputLog",
+        "--repoToken $coverallsToken",
+        "--commitId $commitId",
+        "--commitBranch master",
+        "--commitAuthor `"$author`"",
+        "--commitEmail $email",
+        "--commitMessage `"$message`""
+    )
 
-    $coverallsParams | % { Write-LogPassThru -Message $_ }
+    $coverallsParams | ForEach-Object { Write-LogPassThru -Message $_ }
 
+    Write-LogPassThru -Message "Uploading to CoverAlls"
     & $coverallsExe """$coverallsParams"""
 
+    $bashScriptInvoker = "$PSScriptRoot\CodecovUploader.cmd"
+    $bashScript = "$PSScriptRoot\CodecovUploader.sh"
+    $cygwinLocation = "$env:SystemDrive\cygwin*"
+
+    if($bashScript)
+    {
+        Remove-Item $bashScript -Force
+    }
+
+    Invoke-RestMethod 'https://codecov.io/bash' -OutFile $bashScript
+    Write-BashInvokerScript -path $bashScriptInvoker
+
+    if(Test-Path $bashScriptInvoker -and $bashScript -and $cygwinLocation)
```

Comment:
The suggested script works for finding if one of them is false, if all is OK it returns null. Instead fixed by using the following:

```
if((Test-Path $bashScriptInvoker) -and
    (Test-Path $bashScript) -and
    (Test-Path $cygwinLocation)
```